docs/user_guide/method/method.rst

```diff
@@ -0,0 +1,2384 @@
+The second spectral dimension, i.e., the FT with respect to :math:`t_2`, derives
+its average frequency, :math:`\overline{\Omega}_2`, from a weighted average of
+the :math:`\hat{E}`, and :math:`\hat{F}` transition frequencies. Much of the
+experimental design and implementation of an NMR method is in identifying the
+desired transition pathways and finding ways to acquire their signals while
+eliminating all undesired transition pathway signals.
+
+While NMR measurements occur in the time domain, **mrsimulator** simulates the
+corresponding multi-dimensional spectra directly in the frequency domain. The
+**Method** object in **mrsimulator** needs only a few details of the NMR pulse
+sequence to generate the spectrum. It mimics the result of the pulse sequence
+given the desired transition pathways and their complex amplitudes and average
+frequencies in each spectroscopic dimension of the dataset. To this end, a
+**Method** object is organized according to the UML diagram below.
+
+.. figure:: ../../_static/MethodUML.*
+    :width: 700
+    :alt: figure
+    :align: center
+
+.. note::
+
+ In UML (Unified Modeling Language) diagrams, each class is represented with a
+ box that contains two compartments. The top compartment has the class's name,
+ and the bottom compartment contains the class's attributes. Default attribute
+ values are shown as assignments. A composition is depicted as a binary
+ association decorated with a filled black diamond. Inheritance is shown as a
+ line with a hollow triangle as an arrowhead.
+
+At the heart of a **Method** object, assigned to its attribute
+``spectral_dimensions``, is an ordered list of :ref:`spectral_dim_api` objects
+in the same order as the time evolution dimensions of the experimental NMR
+sequence. In each **SpectralDimension** object, assigned to the attribute
+``events``, is an ordered list of :ref:`event_api` objects, which are divided
+into three types: (1) :py:meth:`~mrsimulator.method.SpectralEvent`, (2)
+:py:meth:`~mrsimulator.method.DelayEvent`, and (3)
+:py:meth:`~mrsimulator.method.MixingEvent`.  This ordered list of Event objects
+is used to select the desired transition pathways and determine their average
+frequency and complex amplitude in the **SpectralDimension**.
+
+.. warning::
+
+  DelayEvent objects are not available in version 0.7 of **mrsimulator**.
+
+**SpectralEvent** and **DelayEvent** objects define which transitions are
+observed during the event and under which transition-dependent frequency
+contributions they evolve. No coherence transfer among transitions or
+populations occurs in a spectral or delay event. The transition-dependent
+frequency contributions during an Event are selected from a list of
+:ref:`enumeration literals<freq_contrib_api>` and placed in the ``freq_contrib``
+attribute of the event. If ``freq_contrib`` is left unspecified, i.e., the
+value of ``freq_contrib`` is set to ``Null``, a default list holding the
```

Comment:
is set to ``None``

Reply:
need to check UML conventions.    I suppose Null in UML translates to None in Python.

Reply:
Yes I think having `Null` in UML is fine, but when referencing attributes of objects outside of UML, we should stick to Python convention